### PR TITLE
Gp 22 fe dash board

### PR DESCRIPTION
**Pull request by dg-panh**
New features:

- Logic routers
- Layout components

### CODE REVIEW

File: src/common/constant.ts [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
 
 
-export const enum AppPath {
-	Login = '/login',
-	Register = '/register',
-	LandingPage = '/landingPage',
-	Dashboard = '/dashboard',
-}
+export const PageName = {
+	LOGIN: 'Login',
```
Comment (danielNg91) [inline src/common/constant.ts]: use enum or Object.Freeze to prevent runtime change value of PageName

File: src/common/constant.ts [hunk 2]
```diff
@@ -1,19 +1,38 @@
...
+	LANDING_PAGE: 'LandingPage',
+	DASHBOARD: 'Dashboard',
+	GAMES: 'MyProject',
+	USERS: 'Users',
+	CLIENTS: 'Client',
+	PAYMENT: 'PaymentPlan'
+};
+
+export const AppPath = (pageName: string) => {
+	const path = {
```
Comment (danielNg91) [inline src/common/constant.ts]: same as above

File: src/components/Scrollbar/Scrollbar.tsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ import { StyledRootScrollbar, StyledScrollbar } from './styles'
 
 
 type Props = {
-	children: ReactElement,
+	children: Array<ReactElement | any>,
```
Comment (danielNg91) [inline src/components/Scrollbar/Scrollbar.tsx]: why use any here?

File: src/hooks/useResponsive.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+	const isMediaUp = useMediaQuery(theme.breakpoints.up(start));
+
+	const isMediaDown = useMediaQuery(theme.breakpoints.down(start));
+
+	const isMediaBetween = useMediaQuery(theme.breakpoints.between(start, end ?? 'xs'));
+
+	const isMediaOnly = useMediaQuery(theme.breakpoints.only(start));
+
+	if (query === 'up') {
```
Comment (danielNg91) [inline src/hooks/useResponsive.ts:16]: use switch case and map these values of query to enum

File: src/Layout/DashboardLayout/Sidebar/Sidebar.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+const StyledAccount = styled('div')(({ theme }) => ({
+	display: 'flex',
+	alignItems: 'center',
+	padding: theme.spacing(2, 2.5),
+	borderRadius: Number(theme.shape.borderRadius) * 1.5,
+	backgroundColor: alpha(theme.palette.grey[500], 0.12),
+}))
+
+const renderContent = (
```
Comment (danielNg91) [inline src/Layout/DashboardLayout/Sidebar/Sidebar.tsx:25]: component hell. Split to smaller components

**Review by danielNg91 — CHANGES_REQUESTED** [review]

File: src/components/Scrollbar/Scrollbar.tsx (hunk 3 above)
Comment (dg-panh) [reply src/components/Scrollbar/Scrollbar.tsx]: I have issue when try to pass children props into this component but I had fixed it and forgot delete any type

File: src/hooks/useResponsive.ts (hunk 4 above)
Comment (dg-panh) [reply src/hooks/useResponsive.ts:16]: @@ thoi no need nha

File: src/utils/cssStyles.ts [hunk 6]
```diff
@@ -12,7 +12,7 @@ type Props = {
 	endColor ?: string,
 }
 
-export function bgBlur(props: Props) {
```
Comment (danielNg91) [inline src/utils/cssStyles.ts:15]: just recommend, `bgBlur(props?: Props): object` is the same as 2 times Type Assertion

File: src/common/constant.ts [hunk 7]
```diff
@@ -1,19 +1,38 @@
...
+	REGISTER: 'Register',
+	LANDING_PAGE: 'LandingPage',
+	DASHBOARD: 'Dashboard',
+	GAMES: 'MyProject',
+	USERS: 'Users',
+	CLIENTS: 'Client',
+	PAYMENT: 'PaymentPlan'
+});
+
+export const AppPath = (pageName: string) => {
```
Comment (danielNg91) [inline src/common/constant.ts]: use enum PageName{}   
export const AppPath = (pageName: PageName)

Comment (danielNg91) [inline src/common/constant.ts]: why this function use PascalCase?

**Review by danielNg91 — CHANGES_REQUESTED** [review]

File: src/common/constant.ts (hunk 7 above)
Comment (dg-panh) [reply src/common/constant.ts:22]: ủa sao kiu dùng object.freeze cũn dc xong h đổi í hã ~.~

**Review by danielNg91 — APPROVED** [review]

File: src/hooks/useResponsive.ts (hunk 4 above)
Comment (danielNg91) [reply src/hooks/useResponsive.ts:16]: sao ko?? :v

File: src/common/constant.ts [hunk 8]
```diff
@@ -1,20 +1,39 @@
...
+	DASHBOARD = 'Dashboard',
+	GAMES = 'MyProject',
+	USERS = 'Users',
+	CLIENTS = 'Client',
+	PAYMENT = 'PaymentPlan'
 }
 
+export const convertNameToPath = (pageName: PageName) => {
+	const path = {
+		[PageName.LOGIN]: '/login',
```
Comment (danielNg91) [inline src/common/constant.ts]: use global var to optimize performance:
const pathMap = new Map<string, string>(
[
[PageName.REGISTER,  '/register'],
[PageName.LANDING_PAGE: '/landingPage'],
...
]
)
export const convertNameToPath = (pageName: PageName) => pathMap.get(pageName)

**Review by danielNg91 — CHANGES_REQUESTED** [review]

**Review by danielNg91 — APPROVED** [review]

**Pull request merged**